synapse/config/_base.py

```diff
@@ -244,7 +244,7 @@ def read_templates(
 
         # Update the environment with our custom filters
         env.filters.update({"format_ts": _format_ts_filter})
-        if self.public_baseurl:
+        if self.root and self.root.public_baseurl:
```

Comment:
I don't understand why this change is necessary.

Reply:
Oh, by:

> The public_baseurl change was due to self.public_baseurl raising an
AssertionError if we try to call it before it exists, which happens to
be the case for the account validity config code. Note that we override
__getattr__ above.

in the commit message I was referring to this bit, but that probably wasn't too clear.

Essentially when trying to call `read_templates` from `AccountValidityConfig.__init__`, I found that `public_baseurl` had not been set yet, and thus this check would fail. This fix got around the issue, and worked since the account validity code didn't include any MXC URLs. But... of course it could in the future.

I just took another look and realised that the issue is due to calling `AccountValidityConfig.read_templates` instead of `RegistrationConfig.read_templates`. In the former, `public_baseurl` is not set, but in the latter it is. I've pushed 58300ee9b to fix this with a little more explanation. Let me know what you think about the solution.